cmd/nerdctl/run_network_linux_test.go

```diff
@@ -529,3 +529,53 @@ func TestSharedNetworkStack(t *testing.T) {
 	base.Cmd("exec", containerNameJoin, "wget", "-qO-", "http://127.0.0.1:80").
 		AssertOutContains(testutil.NginxAlpineIndexHTMLSnippet)
 }
+
+func TestRunContainerWithMACAddress(t *testing.T) {
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+	networkBridge := "testNetworkBridge" + tID
+	networkMACvlan := "testNetworkMACvlan" + tID
+	networkIPvlan := "testNetworkIPvlan" + tID
+	base.Cmd("network", "create", networkBridge, "--driver", "bridge").AssertOK()
+	base.Cmd("network", "create", networkMACvlan, "--driver", "macvlan").AssertOK()
+	base.Cmd("network", "create", networkIPvlan, "--driver", "ipvlan").AssertOK()
+	t.Cleanup(func() {
+		base.Cmd("network", "rm", networkBridge).AssertOK()
+		base.Cmd("network", "rm", networkMACvlan).AssertOK()
+		base.Cmd("network", "rm", networkIPvlan).AssertOK()
+	})
+	tests := []struct {
+		Network string
+		Ok      bool
+	}{
+		{"host", false},
+		{"none", true},
+		{"container:whatever" + tID, false},
+		{"bridge", true},
+		{networkBridge, true},
+		{networkMACvlan, true},
+		{networkIPvlan, false},
+	}
+	for _, test := range tests {
+		macAddress, err := nettestutil.GenerateMACAddress()
+		if err != nil {
+			t.Errorf("failed to generate MAC address: %s", err)
+		}
+		cmd := base.Cmd("run", "--rm", "--network", test.Network, "--mac-address", macAddress, testutil.CommonImage, "ip", "link", "show")
+		if test.Ok {
+			assertContaines := macAddress
+			// set MAC address for network type none shall not work
+			if test.Network == "none" {
+				assertContaines = "00:00:00:00:00:00"
+			}
+			cmd.AssertOutContains(assertContaines)
```

Comment:
maybe we need to use `nerdctl inspect containerId --format "{{.NetworkSettings.MacAddress}}"` here to get the result

Reply:
![CleanShot 2022-10-05 at 01 48 30@2x](https://user-images.githubusercontent.com/7054676/193890142-96695afb-3cea-40f6-92da-2cbaecc35841.jpg)

Reply:
Thanks. The container should be kept in runtime when inspecting it.